lib/timers/promises.js

```diff
@@ -111,7 +111,86 @@ function setImmediate(value, options = {}) {
       () => signal.removeEventListener('abort', oncancel)) : ret;
 }
 
+async function* setInterval(after, value, options = {}) {
+  if (options == null || typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options',
+      'Object',
+      options);
+  }
+  const { signal, ref = true, throwOnAbort = true } = options;
+  validateAbortSignal(signal, 'options.signal');
+  if (typeof ref !== 'boolean') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options.ref',
+      'boolean',
+      ref);
+  }
+
+  if (typeof throwOnAbort !== 'boolean') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'options.throwOnAbort',
+      'boolean',
+      ref);
+  }
+
+  if (signal?.aborted) {
+    if (throwOnAbort) throw new AbortError();
+    return;
+  }
+
+  let onCancel;
+  let notYielded = 0;
+  let passCallback;
+  let abortCallback;
+  const interval = new Timeout(() => {
+    notYielded++;
+    if (passCallback) {
+      passCallback();
+      passCallback = undefined;
+      abortCallback = undefined;
+    }
+  }, after, undefined, true, true);
+  if (!ref) interval.unref();
+  insert(interval, interval._idleTimeout);
+  if (signal) {
+    onCancel = () => {
+      // eslint-disable-next-line no-undef
+      clearInterval(interval);
+      if (abortCallback) {
+        abortCallback(new AbortError());
+        passCallback = undefined;
+        abortCallback = undefined;
+      }
```

Comment:
Assigning `passCallback` and `abortCallback` seems unnecessary here.
```suggestion
      abortCallback?.(new AbortError());
```

Reply:
the undefinds are there because otherwise we sometimes get an error during the tests because of multiple resolves

Reply:
Could you move those to after the `await new Promise` to avoid the repetition? No strong feeling though, if you prefer as it is now, feel free to disregard, I'd personally find it easier to follow if it were down there.

Reply:
Actually, that's where I originally put it. However, it's incorrect because you could still resolve and reject before the await yields, and get multiple resolves.